Assets/Mirror/Transports/KCP/kcp2k/highlevel/KcpPeer.cs

```diff
@@ -570,8 +605,21 @@ public void RawInput(ArraySegment<byte> segment)
             // byte channel = segment[0]; ArraySegment[i] isn't supported in some older Unity Mono versions
             byte channel = segment.Array[segment.Offset + 0];
 
+            // parse cookie
+            uint messageCookie = BitConverter.ToUInt32(segment.Array, segment.Offset + 1);
+
+            // compare cookie to protect against UDP spoofing.
+            // messages won't have a cookie until after handshake.
+            // so only compare if we are authenticated.
+            // simply drop the message if the cookie doesn't match.
+            if (state == KcpState.Authenticated && messageCookie != cookie)
+            {
+                Log.Warning($"KcpPeer: dropped message with invalid cookie: {messageCookie} expected: {cookie}.");
```

Comment:
Why disclose the cookie here?

Reply:
it's a log message on server, only for debugging.
could make it log.info too.